src/routers/UpdateDemographicsRouter.ts

```diff
@@ -0,0 +1,49 @@
+import { Request } from 'express';
+import User from '../entities/User';
+import CornellMajorRepo from '../repos/CornellMajorRepo';
+import UserRepo from '../repos/UserRepo';
+import AuthenticatedAppplicationRouter from '../utils/AuthenticatedApplicationRouter';
+
+class UpdateDemographicsRouter extends AuthenticatedAppplicationRouter<void> {
+  constructor() {
+    super('POST');
+  }
+
+  getPath(): string {
+    return '/updateDemographics/';
+  }
+
+  async content(req: Request): Promise<void> {
+    const { user, body } = req;
+
+    const validFields = [
+      'graduationYear',
+      'hometown',
+      'major',
+      'profilePictureURL',
+      'pronouns',
+    ];
+
+    /* sanitize fields */
+    for (const key in body) {
+      if (!validFields.includes(key) || key === 'googleID' || key === 'netID') {
```

Comment:
Doesn't first case take care of second&third cases?

Reply:
`googleID` and `netID` would be in `validFields`, but we're explicitly stating that they're immutable here.